dashboard/app/models/regional_partner.rb

```diff
@@ -79,6 +81,41 @@ def future_pd_workshops_organized
     pd_workshops_organized.future
   end
 
+  def summer_workshops_application_state
+    # Now closed.  (Closed date has passed.)
+    if summer_workshops_latest_apps_close_date && summer_workshops_latest_apps_close_date <= Time.zone.now
+      return WORKSHOP_APPLICATION_STATES[:now_closed]
+    # Currently open.  (Not closed, but open date has passed.)
+    elsif summer_workshops_earliest_apps_open_date && summer_workshops_earliest_apps_open_date <= Time.zone.now
+      return WORKSHOP_APPLICATION_STATES[:currently_open]
+    # Applications open at a known date.  (Not closed, not open, but we do have an opening date in the future.)
+    elsif summer_workshops_earliest_apps_open_date && summer_workshops_earliest_apps_open_date > Time.zone.now
+      return WORKSHOP_APPLICATION_STATES[:opening_at]
+    # Applications open, but not sure when.  (Not closed, not open, but we have no opening date yet.)
+    else
+      return WORKSHOP_APPLICATION_STATES[:opening_sometime]
+    end
+  end
```

Comment:
This is pretty straightforward but could probably benefit from a few unit tests.

Reply:
Agreed.  Planning to add tests soon in a followup.

Reply:
See https://github.com/code-dot-org/code-dot-org/pull/25042